### ISSUE TITLE
Is there any alignment files to download?

### ISSUE THREAD

**Issue by Zhang690683220** [aqlaboratory/openfold#113]
Hi,

We're trying to reproduce the training process. However, the alignment seems to take extremely long time.

We used 128 nodes to align 128 mmcif files (1 file on each node), but it took 13 hours to finish the entire job.

I'm wondering if there is tar file that already aligned all mmcif files for us to download which will helps a lot.

Thanks

**Comment by gahdritz** [comment]
There will be approximately one week from now, when we release our full training data. Stay tuned.

**Comment by llwx593** [comment]
Hi,
I want to know whether the alignment files of full training data sets have been published. 
I don't seem to have found them.
Thanks

**Comment by gahdritz** [comment]
Yes they have. See the RODA link in the README.

**Comment by llwx593** [comment]
Hi,
Thank you very much for the training data。But I have some following questions：

1. RODA contains two dir, one is the training data after PDB dataset alignment, and the other is the training data after uniclust30 self distillation dataset alignment. Don't know if I understand it correctly.

2. Whether the training data after PDB dataset alignment is same with  pdb_mmcif/mmcif_ files through scripts/precompute_ alignments.py processing? If so, should I set mmcif_dir to pdb_mmcif/mmcif_files，alignment_dir to RODA_PATH/pdb/，template_mmcif_dir to pdb_mmcif/mmcif_files？
Thanks

**Comment by llwx593** [comment]
Thank you for your answer. I try to run the training scripts using above method, but I got the “StopIteration Eexception”. The following figure is the location of the exception(in openfold/data/data_modules.py):
![3`@B3DI_LYSDUVS1OTI{IFH](https://user-images.githubusercontent.com/33170085/180180545-d5271d21-6d34-4d6a-bfe6-09a7f04ea119.png)
The value of "flag1" just become 1, then the exception throw. I already check the RODA_PATH/pdb/ is normal. I don't know what else can cause this error.

**Comment by gahdritz** [comment]
Could you print out "self.probabilities" for me?

**Comment by llwx593** [comment]
The value of "self.probabilities" is 1.
![image](https://user-images.githubusercontent.com/33170085/180339476-78c485bc-7717-4e5e-a944-c442d08e22d6.png)
The file structure of the RODA data is:
RODAPATH
--pdb
----101m_A
------a3m
--------bfd_uniclust_hits.a3m, mgnify_hits.a3m, uniref90_hits.a3
------hhr
--------pdb70_hits.hhr
----uniclust30

**Comment by llwx593** [comment]
I found that some chain in RODAPATH/pdb does not exist in pdb_mmcif/mmcif_files/. This will lead to keyerror when query cache with chain id of RODAPATH/pdb. Maybe my mmcif_files/ is different from yours. Can I simply delete these nonexistent chain? I'll try to see if it can be trained normally. If it can, will it affect the accuracy？
Thanks.

**Comment by gahdritz** [comment]
I see now. Since the RODA data is supposed to be generally applicable, it has a slightly different format than that expected by the OF dataloaders. For OF's sake, you should flatten the intermediate `a3m` and `hhr` directories, putting all `.a3m` and `.hhr` files directly in directories corresponding to the individual chains. So e.g.

alignment_dir/
---101m_A
------msa_1.a3m
------msa_2.a3m
------template_hits.hhr
---next_chain
------ etc.

If you also want to use the distillation set in uniclust30/, you should similarly flatten the file format directories.

**Comment by NZ99** [comment]
I'm also encountering this issue. Line 383, which is now the same line as what (in @llwx593's screenshot) was line 380. StopIteration exception.

My mmcif_dir looks like this:
`100d.cif
101d.cif
101m.cif
102d.cif
102l.cif
[...]
9rsa.cif
9rub.cif
9wga.cif
9xia.cif
9xim.cif`

My alignment_dir looks like this:
`101m_A
102l_A
102m_A
103l_A
103m_A
[...]
9msi_A
9pai_A
9pai_B
9pcy_A
9rub_A`
and, for example `101m_A`'s directory looks like this:
`bfd_uniclust_hits.a3m  mgnify_hits.a3m  pdb70_hits.hhr  uniref90_hits.a3m`

My chain_data_cache.json looks like this:

> {
>     "2v8c_A": {
>         "release_date": "2007-12-18",
>         "seq": "MAGWQSYVDNLMCDGCCQEAAIVGYCDAKYVWAATAGGVFQSITPVEIDMIVGKDREGFFTNGLTLGAKKCSVIRDSLYVDGDCTMDIRTKSQGGEPTYNVAVGRAGRVLVFVMGKEGVHGGGLNKKAYSMAKYLRDSGF",
>         "resolution": 1.98,
>         "cluster_size": -1
>     },
>     [next chain]
> 

and my mmcif_cache.json looks like this:

> {
>     "5d75": {
>         "release_date": "2016-04-06",
>         "chain_ids": [
>             "A"
>         ],
>         "seqs": [
>             "PKYTKSVLKKGDKTNFPKKGDVVHCWYTGTLQDGTVFDTNIQTSAKKKKNAKPLSFKVGVGKVIRGWDEALLTMSKGEKARLEIEPEWAYGKKGQPDAKIPPNAKLTFEVELVDIDLEHHHHHH"
>         ],
>         "no_chains": 1,
>         "resolution": 1.83
>     },
>     [next chain]

I'm trying to train OpenFold using the RODA-provided alignments, on a SLURM cluster with 4 nodes each with 8 A100s. This error also pops up with a single node with 8 A100s. The command is the following:
`python3 /fsx/openbioml/openfold/train_openfold.py /fsx/openbioml/openfold_data/pdb_mmcif/mmcif_files/ /fsx/openbioml/openfold_roda/pdb/ /fsx/openbioml/openfold_data/pdb_mmcif/mmcif_files /fsx/openbioml/openfold_out/ 2021-10-10 --template_release_dates_cache_path /fsx/openbioml/openfold_data/mmcif_cache.json --precision 16 --num_nodes 1 --gpus 8 --replace_sampler_ddp=True --seed 42 --deepspeed_config_path /fsx/openbioml/openfold/deepspeed_config.json --checkpoint_every_epoch --train_chain_data_cache_path /fsx/openbioml/openfold_data/chain_data_cache.json --obsolete_pdbs_file_path /fsx/openbioml/openfold_data/pdb_mmcif/obsolete.dat`

This is all on the latest commit. Do you have any recommendation? I'm struggling to fix it.

**Comment by llwx593** [comment]
Thanks for your reply. I tried to skip the nonexistent Chain ID and found that it could train normally, even if I didn't flatten the data.
![image](https://user-images.githubusercontent.com/33170085/181663512-ba4287e7-3bf7-4cc5-ab98-d8d35cfcae7e.png)

**Comment by gahdritz** [comment]
It's important that you flatten the data, or the model is going to run with empty MSAs and templates. It doesn't know how to read un-flattened data like you have.

**Comment by llwx593** [comment]
Oh, maybe I run with empty MSAs and templates. I will try to flatten the data. Thanks.
